libgeopmd/src/DerivativeSignal.cpp

```diff
@@ -100,14 +101,14 @@ namespace geopm
             throw Exception("setup_batch() must be called before sample().",
                             GEOPM_ERROR_RUNTIME, __FILE__, __LINE__);
         }
-        bool is_sampled = m_time_sig->is_sampled();
         double time = m_time_sig->sample();
         size_t history_size = m_history.size();
         // Check if this is the first call ever to sample() (history_size == 0)
         // Or check if this is the first call to sample() since the last call to read_batch()
-        if (history_size == 0ULL || ! is_sampled) {
+        if (history_size == 0ULL || m_last_time != time) {
```

Comment:
Test with GPU ACTIVITY. IIRC, timestamp doesn't update until there's activity on the GPU. So NAN might persist until there's activity on GPU, which causes issues on report, as the average will not reflect NAN samples.

Reply:
I have put a fix in place specifically for the activity timestamps.  Would be great to test the CA agents with these changes.